### ISSUE TITLE
Qgrid install breaks bqplot in jupyterlab

### ISSUE THREAD

**Issue by timkpaine** [quantopian/qgrid#154]
bootstrap 0e08c6e8d1d771f76828:103 GET http://localhost:8888/nullnbextensions/qgrid/0.bundle.js net::ERR_ABORTED
requireEnsure @ bootstrap 0e08c6e8d1d771f76828:103
(anonymous) @ plugin.js:95
exports @ plugin.js:94
WidgetManager.loadClass @ manager.js:143
(anonymous) @ manager-base.js:262
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase._make_model @ manager-base.js:256
(anonymous) @ manager-base.js:246
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase.new_model @ manager-base.js:232
ManagerBase.handle_comm_open @ manager-base.js:144
(anonymous) @ manager.js:82
(anonymous) @ default.js:901
Promise resolved (async)
DefaultKernel._handleCommOpen @ default.js:897
DefaultKernel._onWSMessage @ default.js:729
_ws.onmessage @ default.js:653
manager-base.js:272 Could not instantiate widget
(anonymous) @ manager-base.js:272
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
rejected @ manager-base.js:15
Promise rejected (async)
step @ manager-base.js:16
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase._make_model @ manager-base.js:256
(anonymous) @ manager-base.js:246
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase.new_model @ manager-base.js:232
ManagerBase.handle_comm_open @ manager-base.js:144
(anonymous) @ manager.js:82
(anonymous) @ default.js:901
Promise resolved (async)
DefaultKernel._handleCommOpen @ default.js:897
DefaultKernel._onWSMessage @ default.js:729
_ws.onmessage @ default.js:653
renderer.js:84 Error displaying widget
renderer.js:85 Error: Loading chunk 0 failed.
    at HTMLScriptElement.onScriptComplete (bootstrap 0e08c6e8d1d771f76828:98)
utils.js:119 Error: Could not create a model.
    at promiseRejection (utils.js:119)
    at <anonymous>
promiseRejection @ utils.js:119
Promise rejected (async)
ManagerBase.handle_comm_open @ manager-base.js:149
(anonymous) @ manager.js:82
(anonymous) @ default.js:901
Promise resolved (async)
DefaultKernel._handleCommOpen @ default.js:897
DefaultKernel._onWSMessage @ default.js:729
_ws.onmessage @ default.js:653
bootstrap 0e08c6e8d1d771f76828:98 Uncaught (in promise) Error: Loading chunk 0 failed.
    at HTMLScriptElement.onScriptComplete (bootstrap 0e08c6e8d1d771f76828:98)
onScriptComplete @ bootstrap 0e08c6e8d1d771f76828:98
Promise rejected (async)
ManagerBase.register_model @ manager-base.js:208
(anonymous) @ manager-base.js:247
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase.new_model @ manager-base.js:232
ManagerBase.handle_comm_open @ manager-base.js:144
(anonymous) @ manager.js:82
(anonymous) @ default.js:901
Promise resolved (async)
DefaultKernel._handleCommOpen @ default.js:897
DefaultKernel._onWSMessage @ default.js:729
_ws.onmessage @ default.js:653
bootstrap 0e08c6e8d1d771f76828:98 Uncaught (in promise) Error: Loading chunk 0 failed.
    at HTMLScriptElement.onScriptComplete (bootstrap 0e08c6e8d1d771f76828:98)
onScriptComplete @ bootstrap 0e08c6e8d1d771f76828:98
Promise rejected (async)
ManagerBase.handle_comm_open @ manager-base.js:149
(anonymous) @ manager.js:82
(anonymous) @ default.js:901
Promise resolved (async)
DefaultKernel._handleCommOpen @ default.js:897
DefaultKernel._onWSMessage @ default.js:729
_ws.onmessage @ default.js:653

**Comment by timkpaine** [comment]
https://github.com/bloomberg/bqplot/issues/587

**Comment by jasongrout** [comment]
That URL: ` http://localhost:8888/nullnbextensions/qgrid/0.bundle.js ` has a `null` in it that probably shouldn't be there. Do you have a set of instructions to reproduce this with a clean conda environment?

**Comment by timkpaine** [comment]
```
pip install bqlot qgid==1.0.0b10
jupyter labextension install bqplot
jupyter labextension install qgrid@1.0.0-beta.10
```

Strangely, using the qgrid extension doesn't trigger that error. Only when you try to use bqplot does it try to fetch that url

**Comment by TimShawver** [comment]
Hmm, I haven't been able to reproduce this yet.  I just tested with the following versions and everything seems to be working:
```
# Relevant pip packages:
bqplot==0.10.2
jupyterlab==0.30.5
notebook==5.2.2
qgrid==1.0.0b10

# Installed jupyter lab extensions 
@jupyter-widgets/jupyterlab-manager v0.31.3
bqplot v0.3.2
qgrid v1.0.0-beta.10 
```

Did you try going through the installation steps again in a new virtual env? Might be worth giving that a shot if you haven't yet. 

If that doesn't work could you give me the versions of the various packages that you're using, as I did above? Could you also please include the type of OS you're on and the python version you're using as well? Thanks.

**Comment by timkpaine** [comment]
```
@jupyter-widgets/jupyterlab-manager
        @jupyter-widgets/jupyterlab-manager v0.31.3  enabled  OK
@jupyterlab/plotly-extension
        @jupyterlab/plotly-extension v0.13.0  enabled  OK
bqplot
        bqplot v0.3.1  enabled  OK
jupyterlab_bokeh
        jupyterlab_bokeh v0.1.0  enabled  OK
qgrid
        qgrid v1.0.0-beta.10  enabled  OK


bqplot==0.10.0
jupyterlab==0.30.6
qgrid==1.0.0b10
bokeh==0.12.10
```

**Comment by timkpaine** [comment]
@jasongrout I get 2 bundles too which is weird
```
                           0.bundle.js     281 kB       0  [emitted]  [big]  @jupyter-widgets/controls
                           1.bundle.js     890 kB       1  [emitted]  [big]
                        main.bundle.js    12.5 MB       2  [emitted]  [big]  main
```

**Comment by timkpaine** [comment]
updated to qgrid 1.0.0 and still occurring

**Comment by TimShawver** [comment]
OK, I tried reproducing again using the same versions of the various packages that you're using and I still couldn't reproduce it.

Did you try creating a new virtualenv and installing just qgrid and bqplot to confirm that none of the other packages that you have installed are related? My experience with widgets in general has been that starting from scratch with a new virtual env can clear up lots of problems.

Also where exactly are you seeing the error you pasted? Is it when you run `jupyter labextension install qgrid@1.0.0` or when you attempt to execute a qgrid cell in the notebook?

**Comment by timkpaine** [comment]
it pops up when you go to use bqplot
after installing both (which works just fine), try to do:
```python
import bqplot.pyplot as plt
plt.plot([1,2], [3,4])
plt.show()
```

**Comment by timkpaine** [comment]
```bash
python3 -m virtualenv venv
cd venv
. bin/activate
pip3 install jupyterlab bqplot qgrid
jupyter labextension install @jupyter-widgets/jupyterlab-manager
jupyter labextension install qgrid
jupyter labextension install bqplot
jupyter lab
```

```python
import qgrid
import bqplot.pyplot as plt
import pandas as pd

qgrid.show_grid(pd.DataFrame())

plt.plot([1,2],[3,4])
plt.show()
```

**Comment by TimShawver** [comment]
Gotcha, yep I'm seeing the same behavior. qgrid always works even if bqplot is installed but bqplot always stops working as soon as qgrid is installed (i.e. installation order doesn't matter).  @jasongrout do you have any thoughts on this? I'm afraid I don't know the internals of the jupyterlab widget machinery well enough to diagnose this.

In qgrid I attempted to copy what bqplot was doing in terms of the labextension installation:
https://github.com/quantopian/qgrid/blob/master/js/src/jupyterlab-plugin.js
https://github.com/quantopian/qgrid/blob/master/js/package.json#L49

**Comment by timkpaine** [comment]
They're very complimentary packages so hopefully we can get to the bottom of this. I've been on holiday but now I'll try to dig a bit.

**Comment by jasongrout** [comment]
I think the problem is this line: https://github.com/quantopian/qgrid/blob/84e94f7e0abb7b43ba050399048de9a1fc18fe81/js/src/index.js#L8

That's messing up the jupyterlab webpack machinery. Instead, you should set up the URLs for the classic notebook extension like bqplot does: https://github.com/bloomberg/bqplot/blob/master/js/src/extension.js

**Comment by jasongrout** [comment]
It looks like you already have a corresponding extension.js. It may be enough to just remove the webpack line above. Do you know why it is there?

**Comment by jasongrout** [comment]
Of course, I could have read the comment above it: https://github.com/quantopian/qgrid/blob/84e94f7e0abb7b43ba050399048de9a1fc18fe81/js/src/index.js#L5-L7

Do you know if the line is needed after all? It may be best to put that line in a file that is only used in the classic notebook extension bundle, but not in the jlab code.

**Comment by TimShawver** [comment]
It looks like this came from the widget-cookiecutter repository which I used to get started on this version of qgrid: https://github.com/jupyter-widgets/widget-cookiecutter/blob/master/%7B%7Bcookiecutter.github_project_name%7D%7D/js/lib/index.js#L8

**Comment by TimShawver** [comment]
So I don't really know what it's for, was just attempting to follow convention.  I can try deleting it and see if that fixes things.

**Comment by jasongrout** [comment]
Hmmm, maybe we need to update the cookie cutter...

**Comment by TimShawver** [comment]
That worked @jasongrout, thanks for the help!

@timkpaine I've uploaded a beta version to pypi with the fix and confirmed that it works.  Install as follows:
```
pip install qgrid==1.0.1b0
jupyter labextension install qgrid@1.0.1-beta.0
```
I haven't uploaded conda packages for this version yet...hopefully will do an official release soon so I'm going to wait until I do the official release (since the conda build is painfully manual at the moment).

**Comment by timkpaine** [comment]
@TimShawver awesome, will test in a bit

**Comment by timkpaine** [comment]
@TimShawver confirmed working for me

**Comment by TimShawver** [comment]
This fix was included in version 1.0.1 which has now been released. Thanks again for reporting this.
